### PR TITLE
ENH: add optional pytmc dependency

### PR DESCRIPTION
**Pull request by klauer**
If available, grabs PV name, precision, additional fields, and amplifier flags from pytmc pragmas

TODO
* `pytmc` not yet a test dependency
* The von hamos project in the test suite does not have any pragmas; need to update that project or add a new one

cc @ZLLentz

### CODE REVIEW

File: tcparse/templates/stcmd_default.cmd [hunk 1]
```diff
@@ -57,18 +57,23 @@ asynSetTraceInfoMask("$(ASYN_PORT)", -1, 5)
...
 epicsEnvSet("ECAXISFIELDINIT", "{{motor.additional_fields}}")
 epicsEnvSet("AMPLIFIER_FLAGS", "{{motor.amplifier_flags}}")
+epicsEnvSet("MOTOR_PREFIX",    "{{motor.name[0]}}")
+epicsEnvSet("MOTOR_NAME",      "{{motor.name[1]}}")
 
 EthercatMCCreateAxis("$(MOTOR_PORT)", "$(AXIS_NO)", "$(AMPLIFIER_FLAGS)", "$(AXISCONFIG)")
-dbLoadRecords("EthercatMC.template", "PREFIX=$(PREFIX), MOTOR_NAME=$(MOTOR_NAME), R=$(MOTOR_NAME)-, MOTOR_PORT=$(MOTOR_PORT), ASYN_PORT=$(ASYN_PORT), AXIS_NO=$(AXIS_NO), DESC=$(DESC), PREC=$(PREC) $(ECAXISFIELDINIT)")
-dbLoadRecords("EthercatMCreadback.template", "PREFIX=$(PREFIX), MOTOR_NAME=$(MOTOR_NAME), R=$(MOTOR_NAME)-, MOTOR_PORT=$(MOTOR_PORT), ASYN_PORT=$(ASYN_PORT), AXIS_NO=$(AXIS_NO), DESC=$(DESC), PREC=$(PREC) ")
-dbLoadRecords("EthercatMCdebug.template", "PREFIX=$(PREFIX), MOTOR_NAME=$(MOTOR_NAME), MOTOR_PORT=$(MOTOR_PORT), AXIS_NO=$(AXIS_NO), PREC=3")
+dbLoadRecords("EthercatMC.template", "PREFIX=$(MOTOR_PREFIX), MOTOR_NAME=$(MOTOR_NAME), R=$(MOTOR_NAME)-, MOTOR_PORT=$(MOTOR_PORT), ASYN_PORT=$(ASYN_PORT), AXIS_NO=$(AXIS_NO), DESC=$(DESC), PREC=$(PREC) $(ECAXISFIELDINIT)")
```
Comment (ZLLentz) [inline tcparse/templates/stcmd_default.cmd:70]: Last I remember, the ECAXISFIELDINIT here was breaking my IOC. Has something changed?

File: tcparse/stcmd.py [hunk 2]
```diff
@@ -9,10 +9,19 @@
...
+import sys
 
 import jinja2
 
-from .parse import load_project, Symbol_FB_DriveVirtual
+try:
+    import pytmc
+except ImportError:
+    pytmc = None
+else:
```
Comment (ZLLentz) [inline tcparse/stcmd.py:20]: I always forget about the try/except/else structure

**Review by ZLLentz — COMMENTED** [review]
It's hard for me to follow all of this, but it seems like a good idea

File: tcparse/templates/stcmd_default.cmd (hunk 1 above)
Comment (klauer) [reply tcparse/templates/stcmd_default.cmd:70]: You're correct in remembering - but the issue was that it was unset. It's now set here, and configurable via `pytmc` pragmas.

**Comment by klauer** [comment]
Self-merging for now, in the interest of moving things forward.

**Pull request merged**